src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/code/CodeBlob.java

```diff
@@ -135,27 +199,35 @@ public ImmutableOopMapSet getOopMaps() {
 
 
   // Typing
-  public boolean isBufferBlob()         { return false; }
+  public boolean isBufferBlob()         { return getKind() == BufferKind; }
 
-  public boolean isCompiled()           { return false; }
+  public boolean isNMethod()            { return getKind() == NMethodKind; }
 
-  public boolean isNMethod()            { return false; }
+  public boolean isRuntimeStub()        { return getKind() == RuntimeStubKind; }
 
-  public boolean isRuntimeStub()        { return false; }
+  public boolean isUpcallStub()         { return getKind() == UpcallKind; }
 
-  public boolean isUpcallStub()         { return false; }
+  public boolean isDeoptimizationBlob() { return getKind() == DeoptimizationKind; }
 
-  public boolean isDeoptimizationStub() { return false; }
+  public boolean isUncommonTrapBlob()   {
+    if (!VM.getVM().isServerCompiler()) return false;
```

Comment:
Why is the check needed? Why not just return the value `getKind() == UncommonTrapKind` result below?

Reply:
`UncommonTrapKind` and `ExceptionKind` are not initialized for Client VM because corresponding `CodeBlobKind` values are not defined. See `CodeBlob.initialize()`.
Their not initialized value will be 0 which matches `CodeBlobKind::None` value. Returning true in such case will be incorrect.

Reply:
Ok. Leaving UncommonTrapKind and ExceptionKind uninitialized seems a bit error prone. Perhaps they can be given some sort of INVALID value.

Reply:
Done. Initialized them to `Number_Of_Kinds + 1`.